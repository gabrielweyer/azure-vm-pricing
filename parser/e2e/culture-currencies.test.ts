const child_process = require('child_process');
const fs = require('fs');
const readline = require('readline');

describe('End-to-end tests for supported cultures and currencies', () => {
  describe('en-us - English (US)', () => {
    const culture = 'en-us';

    test('USD - US Dollar ($)', (done) => {
      const currency = 'usd';

      assert(done, culture, currency);
    });
  });

  describe('cs-cz - Čeština', () => {
    const culture = 'cs-cz';

    test('EUR - Euro (€)', (done) => {
      const currency = 'eur';

      assert(done, culture, currency);
    });
  });

  describe('da-dk - Dansk', () => {
    const culture = 'da-dk';

    test('DKK - Danish Krone (kr)', (done) => {
      const currency = 'dkk';

      assert(done, culture, currency);
    });
  });

  describe('de-de - Deutsch', () => {
    const culture = 'de-de';

    test('EUR - Euro (€)', (done) => {
      const currency = 'eur';

      assert(done, culture, currency);
    });

    test('CHF - Swiss Franc. (chf)', (done) => {
      const currency = 'chf';

      assert(done, culture, currency);
    });
  });

  describe('en-au - English (Australia)', () => {
    const culture = 'en-au';

    test('AUD - Australian Dollar ($)', (done) => {
      const currency = 'aud';

      assert(done, culture, currency);
    });
  });

  describe('en-ca - English (Canada)', () => {
    const culture = 'en-ca';

    test('CAD - Canadian Dollar ($)', (done) => {
      const currency = 'cad';

      assert(done, culture, currency);
    });
  });

  describe('en-in - English (India)', () => {
    const culture = 'en-in';

    test('INR - Indian Rupee (₹)', (done) => {
      const currency = 'inr';

      assert(done, culture, currency);
    });
  });

  describe('en-gb - English (UK)', () => {
    const culture = 'en-gb';

    test('GBP - British Pound (£)', (done) => {
      const currency = 'gbp';

      assert(done, culture, currency);
    });

    test('NZD - New Zealand Dollar ($)', (done) => {
      const currency = 'nzd';

      assert(done, culture, currency);
    });
  });

  describe('es-es - Español', () => {
    const culture = 'es-es';

    test('EUR - Euro (€)', (done) => {
      const currency = 'eur';

      assert(done, culture, currency);
    });
  });

  describe('es-mx - Español (MX)', () => {
    const culture = 'es-mx';

    test('USD - US Dollar ($)', (done) => {
      const currency = 'usd';

      assert(done, culture, currency);
    });
  });

  describe('fr-fr - Français', () => {
    const culture = 'fr-fr';

    test('EUR - Euro (€)', (done) => {
      const currency = 'eur';

      assert(done, culture, currency);
    });

    test('CHF - Swiss Franc. (chf)', (done) => {
      const currency = 'chf';

      assert(done, culture, currency);
    });
  });

  describe('fr-ca - Français (Canada)', () => {
    const culture = 'fr-ca';

    test('CAD - Canadian Dollar ($)', (done) => {
      const currency = 'cad';

      assert(done, culture, currency);
    });
  });

  describe('it-it - Italiano', () => {
    const culture = 'it-it';

    test('EUR - Euro (€)', (done) => {
      const currency = 'eur';

      assert(done, culture, currency);
    });

    test('CHF - Swiss Franc. (chf)', (done) => {
      const currency = 'chf';

      assert(done, culture, currency);
    });
  });

  describe('hu-hu - Magyar', () => {
    const culture = 'hu-hu';

    test('EUR - Euro (€)', (done) => {
      const currency = 'eur';

      assert(done, culture, currency);
    });
  });

  describe('nb-no - Norsk', () => {
    const culture = 'nb-no';

    test('NOK - Norwegian Krone (kr)', (done) => {
      const currency = 'nok';

      assert(done, culture, currency);
    });
  });

  describe('nl-nl - Nederlands', () => {
    const culture = 'nl-nl';

    test('EUR - Euro (€)', (done) => {
      const currency = 'eur';

      assert(done, culture, currency);
    });
  });

  describe('pl-pl - Polski', () => {
    const culture = 'pl-pl';

    test('EUR - Euro (€)', (done) => {
      const currency = 'eur';

      assert(done, culture, currency);
    });
  });

  describe('pt-br - Português (Brasil)', () => {
    const culture = 'pt-br';

    test('BRL - Brazilian Real (R$)', (done) => {
      const currency = 'brl';

      assert(done, culture, currency);
    });
  });

  describe('pt-pt - Português', () => {
    const culture = 'pt-pt';

    test('EUR - Euro (€)', (done) => {
      const currency = 'eur';

      assert(done, culture, currency);
    });
  });

  describe('sv-se - Svenska', () => {
    const culture = 'sv-se';

    test('SEK - Swedish Krona (kr)', (done) => {
      const currency = 'sek';

      assert(done, culture, currency);
    });
  });

  describe('tr-tr - Türkçe', () => {
    const culture = 'tr-tr';

    test('USD - US Dollar ($)', (done) => {
      const currency = 'usd';

      assert(done, culture, currency);
    });
  });

  describe('ru-ru - Pусский', () => {
    const culture = 'ru-ru';

    test('RUB - Russian Ruble (руб)', (done) => {
      const currency = 'rub';

      assert(done, culture, currency);
    });
  });

  describe('ja-jp - 日本語', () => {
    const culture = 'ja-jp';

    test('JPY - Japanese Yen (¥)', (done) => {
      const currency = 'jpy';

      assert(done, culture, currency);
    });
  });

  describe('ko-kr - 한국어', () => {
    const culture = 'ko-kr';

    test('KRW - Korean Won (₩)', (done) => {
      const currency = 'krw';

      assert(done, culture, currency);
    });
  });

  describe('zh-tw - 中文(繁體)', () => {
    const culture = 'zh-tw';

    test('TWD - Taiwanese Dollar (NT$)', (done) => {
      const currency = 'twd';

      assert(done, culture, currency);
    });
  });
});

function assert(
  done: jest.DoneCallback,
  culture: string,
  currency: string
): void {
  const operatingSystem = 'windows';
  const region = 'us-west';

  const crawler = child_process.spawn(
    'yarn',
    [
      'crawl',
      '-l',
      culture,
      '-c',
      currency,
      '-o',
      operatingSystem,
      '-r',
      region,
      '--debug'
    ],
    { shell: true }
  );

  let crawlerErrors = [];

  crawler.stderr.on('data', (data) => {
    crawlerErrors.push(data);
  });

  crawler.on('close', (code) => {
    try
    {
      expect(code).toBe(0);

      const fileStream = fs.createReadStream(
        `./out/vm-pricing_${region}_${operatingSystem}.csv`
      );
      const rl = readline.createInterface({
        input: fileStream,
        crlfDelay: Infinity
      });

      let foundExpectedVirtualMachine = false;

      rl.on('close', () => {
        if (!foundExpectedVirtualMachine) {
          done(new Error('Did not find "D2 v3" virtual machine.'));
        }
      });

      rl.on('line', (line) => {
        try {
          if (line.startsWith('D2 v3,')) {
            foundExpectedVirtualMachine = true;
            rl.close();
            expect(line).toMatchSnapshot();
            done();
          }
        } catch (error) {
          done(error);
        }
      });
    }
    catch (error)
    {
      done(error);
    }
  });
}
