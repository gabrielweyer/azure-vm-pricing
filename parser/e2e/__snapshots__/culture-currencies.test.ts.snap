// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`End-to-end tests for supported cultures and currencies cs-cz - Čeština EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,217\\",\\"0,1215\\",\\"0,2007\\",\\"0,1052\\",\\"0,1786\\",\\"0,083\\",\\"0,0425\\",\\"0,0159\\""`;

exports[`End-to-end tests for supported cultures and currencies da-dk - Dansk DKK - Danish Krone (kr) 1`] = `"D2 v3,2,8,\\"1,6138\\",\\"0,9034\\",\\"1,4928\\",\\"0,7824\\",\\"1,3279\\",\\"0,6175\\",\\"0,3163\\",\\"0,1181\\""`;

exports[`End-to-end tests for supported cultures and currencies de-de - Deutsch CHF - Swiss Franc. (chf) 1`] = `"D2 v3,2,8,\\"0,2051\\",\\"0,1148\\",\\"0,1897\\",\\"0,0995\\",\\"0,1688\\",\\"0,0785\\",\\"0,0402\\",\\"0,015\\""`;

exports[`End-to-end tests for supported cultures and currencies de-de - Deutsch EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,217\\",\\"0,1215\\",\\"0,2007\\",\\"0,1052\\",\\"0,1786\\",\\"0,083\\",\\"0,0425\\",\\"0,0159\\""`;

exports[`End-to-end tests for supported cultures and currencies en-au - English (Australia) AUD - Australian Dollar ($) 1`] = `"D2 v3,2,8,0.3228,0.1807,0.2986,0.1565,0.2656,0.1235,0.0633,0.0236"`;

exports[`End-to-end tests for supported cultures and currencies en-ca - English (Canada) CAD - Canadian Dollar ($) 1`] = `"D2 v3,2,8,0.286,0.1601,0.2646,0.1387,0.2353,0.1094,0.0561,0.0209"`;

exports[`End-to-end tests for supported cultures and currencies en-gb - English (UK) GBP - British Pound (£) 1`] = `"D2 v3,2,8,0.1945,0.1089,0.1799,0.0943,0.16,0.0744,0.0381,0.0142"`;

exports[`End-to-end tests for supported cultures and currencies en-gb - English (UK) NZD - New Zealand Dollar ($) 1`] = `"D2 v3,2,8,0.369,0.2066,0.3414,0.1789,0.3036,0.1412,0.0723,0.027"`;

exports[`End-to-end tests for supported cultures and currencies en-in - English (India) INR - Indian Rupee (₹) 1`] = `"D2 v3,2,8,17.1264,9.5875,15.8423,8.3034,14.092,6.5531,3.3568,1.2531"`;

exports[`End-to-end tests for supported cultures and currencies en-us - English (US) USD - US Dollar ($) 1`] = `"D2 v3,2,8,0.209,0.117,0.1933,0.1013,0.172,0.08,0.041,0.0153"`;

exports[`End-to-end tests for supported cultures and currencies es-es - Español EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,217\\",\\"0,1215\\",\\"0,2007\\",\\"0,1052\\",\\"0,1786\\",\\"0,083\\",\\"0,0425\\",\\"0,0159\\""`;

exports[`End-to-end tests for supported cultures and currencies es-mx - Español (MX) USD - US Dollar ($) 1`] = `"D2 v3,2,8,0.209,0.117,0.1933,0.1013,0.172,0.08,0.041,0.0153"`;

exports[`End-to-end tests for supported cultures and currencies fr-ca - Français (Canada) CAD - Canadian Dollar ($) 1`] = `"D2 v3,2,8,\\"0,286\\",\\"0,1601\\",\\"0,2646\\",\\"0,1387\\",\\"0,2353\\",\\"0,1094\\",\\"0,0561\\",\\"0,0209\\""`;

exports[`End-to-end tests for supported cultures and currencies fr-fr - Français CHF - Swiss Franc. (chf) 1`] = `"D2 v3,2,8,\\"0,2051\\",\\"0,1148\\",\\"0,1897\\",\\"0,0995\\",\\"0,1688\\",\\"0,0785\\",\\"0,0402\\",\\"0,015\\""`;

exports[`End-to-end tests for supported cultures and currencies fr-fr - Français EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,217\\",\\"0,1215\\",\\"0,2007\\",\\"0,1052\\",\\"0,1786\\",\\"0,083\\",\\"0,0425\\",\\"0,0159\\""`;

exports[`End-to-end tests for supported cultures and currencies hu-hu - Magyar EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,217\\",\\"0,1215\\",\\"0,2007\\",\\"0,1052\\",\\"0,1786\\",\\"0,083\\",\\"0,0425\\",\\"0,0159\\""`;

exports[`End-to-end tests for supported cultures and currencies it-it - Italiano CHF - Swiss Franc. (chf) 1`] = `"D2 v3,2,8,\\"0,2051\\",\\"0,1148\\",\\"0,1897\\",\\"0,0995\\",\\"0,1688\\",\\"0,0785\\",\\"0,0402\\",\\"0,015\\""`;

exports[`End-to-end tests for supported cultures and currencies it-it - Italiano EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,217\\",\\"0,1215\\",\\"0,2007\\",\\"0,1052\\",\\"0,1786\\",\\"0,083\\",\\"0,0425\\",\\"0,0159\\""`;

exports[`End-to-end tests for supported cultures and currencies ja-jp - 日本語 JPY - Japanese Yen (¥) 1`] = `"D2 v3,2,8,30.2183,16.9164,27.9526,14.6508,24.8643,11.5625,5.9228,2.211"`;

exports[`End-to-end tests for supported cultures and currencies ko-kr - 한국어 KRW - Korean Won (₩) 1`] = `"D2 v3,2,8,300.9495,168.4741,278.3855,145.9101,247.6282,115.1528,58.9861,22.0197"`;

exports[`End-to-end tests for supported cultures and currencies nb-no - Norsk NOK - Norwegian Krone (kr) 1`] = `"D2 v3,2,8,\\"2,2581\\",\\"1,2641\\",\\"2,0888\\",\\"1,0948\\",\\"1,858\\",\\"0,864\\",\\"0,4426\\",\\"0,1652\\""`;

exports[`End-to-end tests for supported cultures and currencies nl-nl - Nederlands EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,217\\",\\"0,1215\\",\\"0,2007\\",\\"0,1052\\",\\"0,1786\\",\\"0,083\\",\\"0,0425\\",\\"0,0159\\""`;

exports[`End-to-end tests for supported cultures and currencies pl-pl - Polski EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,217\\",\\"0,1215\\",\\"0,2007\\",\\"0,1052\\",\\"0,1786\\",\\"0,083\\",\\"0,0425\\",\\"0,0159\\""`;

exports[`End-to-end tests for supported cultures and currencies pt-br - Português (Brasil) BRL - Brazilian Real (R$) 1`] = `"D2 v3,2,8,\\"1,1212\\",\\"0,6277\\",\\"1,0372\\",\\"0,5436\\",\\"0,9226\\",\\"0,429\\",\\"0,2198\\",\\"0,082\\""`;

exports[`End-to-end tests for supported cultures and currencies pt-pt - Português EUR - Euro (€) 1`] = `"D2 v3,2,8,\\"0,217\\",\\"0,1215\\",\\"0,2007\\",\\"0,1052\\",\\"0,1786\\",\\"0,083\\",\\"0,0425\\",\\"0,0159\\""`;

exports[`End-to-end tests for supported cultures and currencies ru-ru - Pусский RUB - Russian Ruble (руб) 1`] = `"D2 v3,2,8,\\"12,1742\\",\\"6,8152\\",\\"11,2615\\",\\"5,9025\\",\\"10,0173\\",\\"4,6583\\",\\"2,3862\\",\\"0,8908\\""`;

exports[`End-to-end tests for supported cultures and currencies sv-se - Svenska SEK - Swedish Krona (kr) 1`] = `"D2 v3,2,8,\\"2,3646\\",\\"1,3237\\",\\"2,1873\\",\\"1,1465\\",\\"1,9457\\",\\"0,9048\\",\\"0,4635\\",\\"0,173\\""`;

exports[`End-to-end tests for supported cultures and currencies tr-tr - Türkçe USD - US Dollar ($) 1`] = `"D2 v3,2,8,\\"0,209\\",\\"0,117\\",\\"0,1933\\",\\"0,1013\\",\\"0,172\\",\\"0,08\\",\\"0,041\\",\\"0,0153\\""`;

exports[`End-to-end tests for supported cultures and currencies zh-tw - 中文(繁體) TWD - Taiwanese Dollar (NT$) 1`] = `"D2 v3,2,8,6.6629,3.73,6.1634,3.2304,5.4824,2.5494,1.3059,0.4875"`;
